WordPress/Classes/Services/PostService.m

```diff
@@ -356,8 +415,7 @@ - (void)autoSave:(AbstractPost *)post
                      success(post, nil);
                  } failure:failure];
     } else {
-        NSDictionary *userInfo = @{ NSLocalizedDescriptionKey : @"Previews are unavailable for this kind of post." };
-        failure([NSError errorWithDomain:PostServiceErrorDomain code:0 userInfo:userInfo]);
+        failure(defaultError);
```

Comment:
Using an self-hosted site will end up in this call. However, this doesn't mark the post as failed which produces the bug of the "infinite uploading post...".

We need to mark the post as failed for each failure. :)

Reply:
(╯°□°)╯︵ ┻━┻

Fixed in 928c4aae28fa5d7b230def8d4498cca694c4f4ad.